apps/namadillo/src/App/Masp/ShieldedSyncProgress.tsx

```diff
@@ -0,0 +1,44 @@
+import { shieldedBalanceAtom, shieldedSyncProgress } from "atoms/balance/atoms";
+import { useAtomValue } from "jotai";
+import { useEffect, useMemo, useState } from "react";
+import { twMerge } from "tailwind-merge";
+
+export const ShieldedSyncProgress = (): JSX.Element => {
+  const syncProgress = useAtomValue(shieldedSyncProgress);
+  const { isFetching } = useAtomValue(shieldedBalanceAtom);
+  const [showShieldedSync, setShowShieldedSync] = useState(false);
+
+  const roundedProgress = useMemo(() => {
+    // Only update when the progress changes by at least 1%
+    return Math.min(Math.floor(syncProgress * 100), 100);
+  }, [Math.floor(syncProgress * 100)]);
+
+  useEffect(() => {
+    let timeout: ReturnType<typeof setTimeout> | undefined;
```

Comment:
It looks like it shows the nodejs specific return type because of the node types that we include. Not sure if we need those but it's ok for now :)